Eigen/src/Core/arch/GPU/Half.h

```diff
@@ -485,8 +520,9 @@ EIGEN_STRONG_INLINE EIGEN_DEVICE_FUNC half expm1(const half& a) {
   return half(numext::expm1(float(a)));
 }
 EIGEN_STRONG_INLINE EIGEN_DEVICE_FUNC half log(const half& a) {
-#if defined(EIGEN_HAS_HIP_FP16) && defined(EIGEN_HIP_DEVICE_COMPILE)
-  return half(hlog(a));
+#if (defined(EIGEN_HAS_CUDA_FP16) && EIGEN_CUDACC_VER >= 80000 && defined(EIGEN_CUDA_ARCH) && EIGEN_CUDA_ARCH >= 530) || \
```

Comment:
I'm wondering how do we derive the required version for `EIGEN_CUDACC_VER` and `EIGEN_CUDA_ARCH`?

Reply:
we (as in HIPCC) don't ... we (as in HIPCC) relies on the `||` part on line 524

we (as in EIGEN) defines those macros in `Eigen/Core`

Reply:
I mean how are those values fetched? from existing CUDA implementation of Eigen?

Reply:
from macros implicitly defined within the CUDA compiler...see code here

https://github.com/ROCmSoftwarePlatform/eigen-upstream/blob/PR_fixes/Eigen/Core#L37

and here

https://github.com/ROCmSoftwarePlatform/eigen-upstream/blob/PR_fixes/Eigen/Core#L21